### DISCUSSION TITLE
Authenticating user from 3rd-party cookie

### DISCUSSION THREAD

**Discussion by danielbuechele** [directus/directus#12471] [Archive]
From my 3rd-party authentication service I have a cookie set for `*.mydomain.com`. Directus is running on `directus.mydomain.com`. Now I want to use this cookie to authenticate users in directus, so that users are automatically logged in.

I found the `authenticate` filter hook, where I have access to my cookie and can set the `accountability` object accordingly. However, even after setting the `accountability` object directus isn't considering my user to be authenticated. Is there a way to mark my user as authenticated from a hook?

Are there any other ideas how I can turn my cookie into a proper directus authentication?

**Comment by rijkvanzanten** [comment]
What do you mean by 

> However, even after setting the accountability object directus isn't considering my user to be authenticated.

exactly? Are you still getting permissions errors etc? 🤔

**Reply by danielbuechele** [reply]
Sorry, should have been clearer here. I am getting redirected to the login page of the app. I think that is because before even considering the accountability it is checking for a token or something.

**Reply by rijkvanzanten** [reply]
Ohhh you're trying to log them into the App?

**Reply by danielbuechele** [reply]
Yeah, sorry, should have mentioned that

**Reply by rijkvanzanten** [reply]
Gotcha! In that case, this flow will have to work a little differently. The first time the app opens, it tries hitting the `/refresh` endpoint. That endpoint will check for the Directus cookie, and return a new access token for the app to use. At that point, the app recognizes it has a token and continues to open the app. However, in your case, we'd have to figure out a way to hijack that refresh endpoint to return/create the session based on a different cookie you already have enabled 🤔

**Reply by danielbuechele** [reply]
Is there a way to hook into requests like the `/refresh` endpoint? Then I could hook into this and generate the tokens myself, based on my cookie.

**Reply by rijkvanzanten** [reply]
There's currently no "blocking" `request` hook yet 🤔 I'll gladly accept a PR to add that tho! 🙂

**Reply by danielbuechele** [reply]
While a request hook might be useful for some other use-cases, too, I'm wondering if it's actually the best solution for my problem. The `authenticate` hook is already solving this problem on the API. Would it make sense to add support for this hook on the app as well?

**Reply by rijkvanzanten** [reply]
There's no special endpoints or hooks for the app, as it's just a SPA that uses the regular API endpoints. I think what we'd have to do to allow for this is to fire the authenticate hook on refresh as well, or add a new refresh hook 🙂
